chain/traversal_test.go

```diff
@@ -0,0 +1,154 @@
+package chain_test
+
+import (
+	"context"
+	"testing"
+
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/types"
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
+	"gx/ipfs/QmVmDhyTTUcQXFD1rRQ64fGLMSAoaQvNH3hwuaCFAPq2hy/errors"
+)
+
+func TestGetParentTipSet(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+	require := require.New(t)
+	ctx := context.Background()
+	store := newBlockStore()
+
+	root := store.NewBlock(0)
+	b11 := store.NewBlock(1, root)
+	b12 := store.NewBlock(2, root)
+	b21 := store.NewBlock(3, b11, b12)
+
+	t.Run("root has empty parent", func(t *testing.T) {
+		ts := requireTipset(t, root)
+		parent, e := chain.GetParentTipSet(ctx, store, ts)
+		require.NoError(e)
+		assert.Empty(parent)
+	})
+	t.Run("plural tipset", func(t *testing.T) {
+		ts := requireTipset(t, b11, b12)
+		parent, e := chain.GetParentTipSet(ctx, store, ts)
+		require.NoError(e)
+		assert.Equal(requireTipset(t, root), parent)
```

Comment:
there is a tipset.equals which is presumably better than default object comparison(?)